src/sentry/api/serializers/models/release.py

```diff
@@ -197,6 +189,44 @@ def get_attrs(self, item_list, user, *args, **kwargs):
                 .values('release_id').annotate(new_groups=Sum('new_groups'))
                 .values_list('release_id', 'new_groups')
             )
+        return tags, group_counts_by_release
+
+    def __get_release_data_with_environment(self, project, item_list, environment):
+        if project:
+            release_project_envs = ReleaseProjectEnvironment.objects.filter(
+                project=project, release__in=item_list, environment=environment).select_related('release')
+        else:
+            release_project_envs = ReleaseProjectEnvironment.objects.filter(
+                release__in=item_list, environment=environment).select_related('release')
+
+        first_and_last_seen = {}
+        for release_project_env in release_project_envs:
+            first_and_last_seen[release_project_env.release.version] = {
+                'first_seen': release_project_env.first_seen,
+                'last_seen': release_project_env.last_seen,
+            }
```

Comment:
If the release shows up in multiple projects, won't this mean that whatever version is seen last in `release_project_envs` is going to clobber any values that came before it? Shouldn't this do the `min` of `first_seen` and the `max` of `last_seen` for that release like on line 175?